src/app.controller.ts

```diff
@@ -1,12 +1,198 @@
-import { Controller, Get } from '@nestjs/common';
+import {
+  Controller,
+  HttpStatus,
+  Get,
+  Res,
+  Req,
+  Logger,
+  Body,
+  Post,
+} from '@nestjs/common';
 import { AppService } from './app.service';
+import axios from 'axios';
+import { Request, Response } from 'express';
+
+import { PrismaService } from 'src/prisma/prisma.service';
+import { UserService } from 'src/user/user.service';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserDto, LoginDTO } from './dto/user.dto';
 
 @Controller()
 export class AppController {
-  constructor(private readonly appService: AppService) {}
+  private readonly logger = new Logger(AppController.name);
+  constructor(
+    private readonly appService: AppService,
+    private readonly userService: UserService,
+    private readonly prismaService: PrismaService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  @Get('/')
+  async getRoutesInfo() {
+    return {
+      error: 'Nothing on this route',
+      error_description:
+        'Nothing on this route, go to /login -> (username,password,scopes?,resources?), /jwt-verify -> (token), /opaque-verify -> (token), /signup -> (username,password,email,birthdate,gender) instead',
+    };
+  }
+
+  @Post('/login')
+  async loginRoute(
+    @Req() req: Request,
+    @Res() res: Response,
+    @Body() body: LoginDTO,
+  ) {
+    const { username, password, scopes, resource, grant_type } = body;
+    const jwtToken = req.cookies?.jwt;
+    if (!jwtToken && (!username || !password)) {
+      res.status(400).send({
```

Comment:
raise an expection instead of doing like this:
example snippet:
```suggestion
throw new InvalidRequestException({
        error: 'Invalid Credentials',
        error_description: 'username, password missing in body',
});
```